run.c

```diff
@@ -221,15 +233,22 @@ void softmax(float* x, int size) {
     }
 }
 
-void matmul(float* xout, float* x, float* w, int n, int d) {
+void matmul(float *xout, float *x, float *w, int n, int d)
+{
     // W (d,n) @ x (n,) -> xout (d,)
-    for (int i = 0; i < d; i++) {
+#if ACCELERATE_AVAILABLE
+    vDSP_mmul(w, 1, x, 1, xout, 1, d, 1, n);
+#else
+    for (int i = 0; i < d; i++)
+    {
         float val = 0.0f;
-        for (int j = 0; j < n; j++) {
+        for (int j = 0; j < n; j++)
+        {
```

Comment:
Why did you do this?
It would be way less changes if you had kept the original position of curly braces (on 3 places)